api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/IncomingConnectorFactory.java

```diff
@@ -0,0 +1,112 @@
+ * <p>
+ * <p>
+ * The portion of the key that precedes the {@code attribute} is acts as a property prefix that has a common structure
+ * across all {@code channels}.
+ * </p>
+ * <p>
+ * The {@code channel-name} segment in the configuration key corresponds to the name of the channel used in the
+ * {@code Incoming} annotation:
+ *
+ * <pre>
+ * &#64;Incoming("my-channel")
+ * public void consume(String s) {
+ *      // ...
+ * }
+ * </pre>
+ * <p>
+ * The set of attributes depend on the connector and transport layer (for example, bootstrap-servers is Kafka specific).
+ * The {@code type} attribute is mandatory and indicates the fully qualified name of the {@link MessagingProvider}
+ * implementation. It must match the class returned by the {@link Connector} qualifier used on the bean implementation.
+ * This is how a reactive messaging implementation looks for the specific {@link IncomingConnectorFactory} required for
+ * a channel.
+ *
+ * In the previous configuration, the reactive messaging implementation would need to find the
+ * {@link IncomingConnectorFactory} qualified using the {@link Connector} qualifier indicating the value
+ * {@code i.e.m.reactive.messaging.impl.kafka.Kafka} class to create the {@code my-channel} channel. Note that if the
+ * connector cannot be found, the deployment must be failed with a {@link DeploymentException}.
+ * <p>
+ * The {@link #getPublisherBuilder(Config)} is called for every channel that needs to be created. The {@link Config} object
+ * passed to the method contains a subset of the global configuration, and with the prefixes removed. So for the previous
+ * configuration, it would be:
+ * <pre>
+ * bootstrap-services = localhost:9092
+ * topic = my-topic
+ * </pre>
+ * <p>
+ * In this example, if {@code topic} was missing as a configuration property, the Kafka connector would be at liberty to
+ * default to the stream name indicated in the annotation as the Kafka topic. Such connector specific behaviours are
+ * outside the scope of this specification.
+ * <p>
+ * So the connector implementation can retrieve the value with {@link Config#getValue(String, Class)} and
+ * {@link Config#getOptionalValue(String, Class)}.
+ * <p>
+ * If the configuration is invalid, the {@link #getPublisherBuilder(Config)} method must throw an
+ * {@link IllegalArgumentException}, caught by the reactive messaging implementation and failing the deployment by
+ * throwing a {@link DeploymentException} wrapping the exception.
+ * <p>
+ * Note that a Reactive Messaging implementation must support the configuration format described here. Implementations
+ * are free to provide additional support for other approaches.
+ */
+public interface IncomingConnectorFactory {
+
+    /**
+     * Creates a <em>channel</em> for the given configuration. The given configuration {@code type} attribute matches
+     * the {@link Connector} qualifier used on the bean.
```

Comment:
Creates a <em>channel</em> for the given configuration. The channel's configuration of the {@code connector} attribute can be used to override the {@link Connector} qualifier's parameter indicating the {@IncomingConnectorFactory} implementation class.